Assets/Scripts/Framework/TriggerArea/NetworkTriggerAreaBase.cs

```diff
@@ -0,0 +1,102 @@
+using System.Collections.Generic;
+using FishNet.Object;
+using System.Linq;
+using UnityEngine;
+using System;
+
+namespace DerailedDeliveries.Framework.TriggerArea
+{
+    /// <summary>
+    /// The base class for all NetworkTriggerArea classes.
+    /// </summary>
+    [RequireComponent(typeof(Collider))]
+    public abstract class NetworkTriggerAreaBase<T> : NetworkBehaviour where T : Component
+    {
+        private readonly List<T> _colliders = new();
+
+        /// <summary>
+        /// Invoked when a collider enters or exits the trigger area.
+        /// </summary>
+        public Action<T[]> OnColliderChange;
+
+        /// <summary>
+        /// Invoked when a collider enters the trigger area.
+        /// </summary>
+        public Action<T> OnColliderEnter;
+
+        /// <summary>
+        /// Invoked when a collider exits the trigger area.
+        /// </summary>
+        public Action<T> OnColliderExit;
+
+        /// <summary>
+        /// Gets the colliders currently inside the trigger area.
+        /// </summary>
+        public T[] ComponentsInCollider => _colliders.ToArray();
+
+        [SerializeField]
+        private int _framesUntillUpdate = 25;
+
+        private int _framesPassedSinceUpdate = 0;
+
+        private void OnTriggerEnter(Collider other)
+        {
+            if (!other.TryGetComponent(out T component))
+                return;
+
+            AddNewComponent(component);
+        }
+
+        private void OnTriggerExit(Collider other)
+        {
+            if (!other.TryGetComponent(out T component))
+                return;
+
+            RemoveOldComponent(component);
+        }
+
+        private void AddNewComponent(T component)
+        {
+            if (_colliders.Contains(component))
+                return;
+
+            _colliders.Add(component);
+            OnColliderEnter?.Invoke(component);
+            OnColliderChange?.Invoke(_colliders.ToArray());
+        }
+
+        private void RemoveOldComponent(T component)
+        {
+            if (!_colliders.Contains(component))
+                return;
+
+            _colliders.Remove(component);
+            OnColliderExit?.Invoke(component);
+            OnColliderChange?.Invoke(_colliders.ToArray());
+        }
+
+        private protected virtual void FixedUpdate()
+        {
+            if (_framesPassedSinceUpdate % _framesUntillUpdate != 0)
```

Comment:
The mod operator is not needed for this.
```suggestion
            if (_framesPassedSinceUpdate == _framesUntillUpdate)
```